src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -125,7 +126,19 @@ public override bool TryGetVariable<T>(string name, out T value)
                 switch (variable.Value)
                 {
                     case PSObject psobject:
-                        value = (T) psobject.BaseObject;
+                        if (psobject.BaseObject is PSCustomObject)
+                        {
+                            Dictionary<string, object> table = new Dictionary<string, object>();
+                            foreach (var p in psobject.Properties)
+                            {
+                                table.Add(p.Name, p.Value);
+                            }
+                            value = (T) ((Object)table);
```

Comment:
This is likely going to fail, unless `T` is `object`.

Reply:
casting to object is the only safe way to then cast to T, might introduce boxing and unboxing

Reply:
I meant `table` here is of the type `Dictionary<string, object>`, so unless `T` itself is `object` or `Dictionary<string, object>`, the casting will fail at runtime.
But I don't know if this is a real problem, maybe `T` will always be `object` when calling `TryGetVariable`?

Reply:
I agree with Dongbo. For a `TryGetVariable` method, it's probably best to wrap that in a try/catch or use `is`/`as`.

what do you think of this?

```csharp
public override bool TryGetVariable<T>(string name, out T value)
{
    var variable = pwsh.Runspace.SessionStateProxy.PSVariable.Get(name);
    if (variable != null)
    {
        object output;
        switch (variable.Value)
        {
            case PSObject psobject:
                if (psobject.BaseObject is PSCustomObject)
                {
                    Dictionary<string, object> table = new Dictionary<string, object>();
                    foreach (var p in psobject.Properties)
                    {
                        table.Add(p.Name, p.Value);
                    }
                    output = table;
                }
                else
                {
                    output = psobject.BaseObject;
                }
                break;
            default:
                output = variable.Value;
                break;
        }

        if (output is T tval)
        {
            value = tval;
            return true;
        }
    }

    value = default;
    return false;
}

```